src/ProgressBar.tsx

```diff
@@ -18,18 +20,27 @@ const sizeMap = {
   default: '8px'
 }
 
-const ProgressContainer = styled.span`
+const ProgressContainer = styled.span<
+  {
+    inline?: boolean
+    barSize?: keyof typeof sizeMap
+  } & WidthProps &
+    SystemCommonProps &
+    SxProp
+>`
   display: ${props => (props.inline ? 'inline-flex' : 'flex')};
   overflow: hidden;
   background-color: ${get('colors.gray.2')};
   border-radius: ${get('radii.1')};
-  height: ${props => sizeMap[props.barSize]};
+  height: ${props => sizeMap[props.barSize || 'default']};
```

Comment:
Is this necessary if we're setting the default value in defaultProps?

Reply:
It shouldn't be necessary. But defaultProps + styled-components + TypeScript don't play well together in this case. TypeScript was complaining about `props.barSize` possibly being undefined. I'm open to other solutions :)